addons/product/models/product_template.py

```diff
@@ -354,55 +340,37 @@ def _set_barcode(self):
         self._set_product_variant_field('barcode')
 
     @api.model
-    def _get_weight_uom_id_from_ir_config_parameter(self):
+    def _get_weight_uom_name_from_ir_config_parameter(self):
         """ Get the unit of measure to interpret the `weight` field. By default, we considerer
         that weights are expressed in kilograms. Users can configure to express them in pounds
         by adding an ir.config_parameter record with "product.product_weight_in_lbs" as key
         and "1" as value.
         """
         product_weight_in_lbs_param = self.env['ir.config_parameter'].sudo().get_param('product.weight_in_lbs')
-        if product_weight_in_lbs_param == '1':
-            return self.env.ref('uom.product_uom_lb')
-        else:
-            return self.env.ref('uom.product_uom_kgm')
+        return 'lb' if product_weight_in_lbs_param == '1' else 'kg'
```

Comment:
Don't forget to change this back to way it was after undoing the UoM deletions. These UoM names are translated